src/components/AutoCompleteInput/AutoCompleteInput.js

```diff
@@ -68,46 +84,87 @@ const itemBaseStyles = ({ theme }) => css`
   }
 `;
 
-const itemHighlight = ({ selected, theme }) =>
+const optionHighlight = ({ selected, theme }) =>
   selected &&
   css`
     color: ${theme.colors.p500};
   `;
 
-const Item = styled('div')(itemBaseStyles, itemHighlight);
+const Option = styled('div')(optionBaseStyles, optionHighlight);
```

Comment:
What is the benefit of declaring the styles as separate variables and wrapping them this way?

Reply:
We've developed this convention in Circuit UI for 2 reasons:

1. **Style composition**: When an element's styles change based on a prop, there are two common ways to write this in Emotion:

_Inline_

```js
const Text = styled.p`
  font-size: 16px;
  line-height: 1.75;
  font-weight: ${props =>
    props.bold ? props.theme.fontWeight.bold : props.theme.fontWeight.normal};
`;
```

_Composition_

```js
const baseStyles = ({ theme }) => css`
  font-size: 16px;
  line-height: 1.75;
  font-weight: ${props.theme.fontWeight.normal};
`;

const boldStyles = ({ theme, bold }) => bold && css`
  font-weight: ${props.theme.fontWeight.bold};
`;

const Text = styled.p(baseStyles, boldStyles);
```

While the second one is less concise, it drastically improves code readability. You can see at one glance how a certain prop changes the styles. Imagine another prop `active` that changes the `font-weight` as well. In the _inline_ approach, you'd now need a nested ternary or a separate function to determine the value. In the _composition_ approach, you simply add another style object and add it in the list of styles. The order determines which style takes precedence (just like in vanilla CSS).

Another added benefit is that you can share style objects between multiple elements. 

2. **Static style extraction**: Circuit UI has experimental support for [static CSS](https://circuit.sumup.com/?path=/docs/advanced-static-css--page). This is made possible by the `label` CSS property that you see here and in other components. The label is used as the class name when the CSS is extracted.

Does this make it clearer? :)

Reply:
Yes, that's a super helpful explanation. Thanks so much! 😊